src/fontra/views/editor/sidebar.js

```diff
@@ -24,7 +24,7 @@ export class Sidebar {
 
     const panelContent = html.div(
       { "class": "sidebar-content", "data-sidebarName": panel.identifier },
-      [panel]
+      [panel instanceof HTMLElement ? panel : panel.contentElement]
```

Comment:
I'm still not happy with this switch. Can you make the demo plugin work with the original `[panel]` line?

We can also consider change the `addSidebarPanel()` call signature to take three arguments:
1. the config object (containing `iconPath` etc), which may _also_ be the content element
2. the sidebar "name" (left/right)
3. *optionally* a separate content element

How does that sound to you?

Reply:
We discussed about defining the panel in remote plugin declare as a web component in our call. Do you still want that, or you want the second one? Web component in remote panel sounds like a better solution to me. The panels will have their own shadow root in that way. The switch you're not happy with will be deleted.

Reply:
I think it would be nice to support both, which is possible with my suggestion. As your demo shows, it is not always needed nor convenient to write a web component. So yes, please continue to implement my suggestion.

Reply:
Creating something like method signature overloading is tricky in Javascript. I still think about how to make it clearly.

Reply:
I create panel component when this method is called. I am not sure if we should do it in that way but I couldn't find a better solution either.

Reply:
I don't understand. It's not method signature overloading and it is not tricky. There is simply a third argument that is optional. If it is not given, it is equal to undefined. We separate the panel config (id, icon path) from the content element. If the third argument is undefined, we use the first argument for this instead (so one object is both the config _and_ the contents). This may have to propagate to`sidebar.addPanel()` panel as well. There should be no need for trickeries like creating a subclass on the fly.

Am I missing something? Perhaps we should have a call to figure this out.

Reply:
Please read the commit, you will see what I'm talking about.